packages/loader/container-loader/src/container.ts

```diff
@@ -975,7 +978,12 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             // Note: no need to fetch ops as we do it preemptively as part of DeltaManager.attachOpHandler().
             // If there is gap, we will learn about it once connected, but the gap should be small (if any),
             // assuming that connect() is called quickly after initial container boot.
-            this.connectInternal({ reason: "DocumentConnect", fetchOpsFromStorage: false });
+            this._connectAbortController = new AbortController();  // Create a new signal for each connect() call
```

Comment:
Similar feedback as in connection manager: overwriting this._connectAbortController will result in two calls to connect() not to be working correctly.

Reply:
BTW, it would be great to have a test where
1. Two connect() calls are done one after another.
2. cancellation on first IProgress is made and validated that whole process cancells out eventually
3. Same flow, but cancel second IProgress (change in # 2) and test the same.

In both places we need to ensure that both callers can learn about cancelation.

Reply:
@vladsud, do you feel strongly that users should be able to manually cancel a `connect()` function? This would potentially be done by `connect()` returning an `IProgress` object that users could use to cancel the connection attempt. 
I was talking with @markfields, and we thought that this may not be desired. Since the `disconnect()` flow will have to internally cancel any in-progress connection attempts, users would effectively have two ways to cancel a connection attempt. Another smaller reason is that returning `IProgress` from `connect()` would require a breaking change to change the call signature.